BrainPortal/app/controllers/nh_projects_controller.rb

```diff
@@ -87,8 +87,9 @@ def show #:nodoc:
   end
 
   def files #:nodoc:
-    @nh_project = find_nh_project(current_user, params[:id])
-    @files      = @nh_project.userfiles
+    @nh_project      = find_nh_project(current_user, params[:id])
+    available_dp_ids = DataProvider.find_all_accessible_by_user(current_user).pluck(:id)
+    @files           = @nh_project.userfiles.where(:data_provider_id => available_dp_ids)
```

Comment:
@prioux after our discussions do we need to restrict to available_dp_ids ?

Reply:
Nope, not really. I'll fix